### PR TITLE
[Merged by Bors] - feat: port algebra.group.ulift

### PR DESCRIPTION
**Pull request by arienmalec**
Port of Algebra.Group.ULift

based on 655994e298904d7e5bbd1e18c95defd7b543eb94

### CODE REVIEW

File: Mathlib/Algebra/Group/ULift.lean [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+  Equiv.ulift.injective.commGroupWithZero _ rfl rfl (fun _ _ => rfl) (fun _ => rfl) (fun _ _ => rfl)
+    (fun _ _ => rfl) fun _ _ => rfl
+#align ulift.comm_group_with_zero ULift.commGroupWithZero
+
+@[to_additive AddLeftCancelSemigroup]
+instance leftCancelSemigroup [LeftCancelSemigroup α] : LeftCancelSemigroup (ULift α) :=
+  Equiv.ulift.injective.leftCancelSemigroup _ fun _ _ => rfl
+#align ulift.left_cancel_semigroup ULift.leftCancelSemigroup
+
+@[to_additive AddRightCancelSemigroup]
```
Comment (semorrison) [inline Mathlib/Algebra/Group/ULift.lean:188]:
```suggestion
@[to_additive addRightCancelSemigroup]
```
and all the others.

File: Mathlib/Algebra/Group/ULift.lean [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+universe u v
+
+variable {α : Type u} {β : Type _} {x y : ULift.{v} α}
+
+namespace ULift
+
+@[to_additive]
+instance one [One α] : One (ULift α) :=
+  ⟨⟨1⟩⟩
+#align ulift.has_one ULift.one
```
Comment (semorrison) [inline Mathlib/Algebra/Group/ULift.lean:30]: Please duplicate all the #aligns for the additive versions.

**Comment by semorrison** [comment]
bors d+

File: Mathlib/Algebra/Group/ULift.lean (hunk 1 above)
Comment (arienmalec) [reply Mathlib/Algebra/Group/ULift.lean]: just removed manual override...

**Pull request closed without merging**